src/Graph/Trees/DecisionTree.php

```diff
@@ -335,4 +363,68 @@ protected function splitImpurity(array $groups) : float
 
         return $impurity;
     }
+
+    /**
+     * Recursive function to print out the decision rule at each node
+     * using preorder traversal.
+     *
+     * @param string $carry
+     * @param int $nodeCounter
+     * @param int $maxDepth
+     * @param \Rubix\ML\Graph\Nodes\BinaryNode $node
+     * @param ?int $parentId
+     * @param ?int $leftRight
+     * @param int $depth
+     * @param int $nodesCounter
+     */
+    protected function _rules(
+        string &$carry,
+        int &$nodesCounter,
+        int $maxDepth = -1,
+        BinaryNode|Split|Decision $node,
+        ?int $parentId = null,
+        ?int $leftRight = null,
+        int $depth = 0
+    ) : void {
+        ++$depth;
+
+        $thisNode = $nodesCounter++;
+
+        if ($depth == $maxDepth) {
```

Comment:
Very minor but strict comparison here is possible since both `$depth` and `$maxDepth` are integer variables.

Reply:
Makes sense. Will do.